static/app/utils/profiling/flamegraph.ts

```diff
@@ -10,75 +10,72 @@ export class Flamegraph {
   profile: Profile;
   frames: FlamegraphFrame[] = [];
 
-  name: string;
   profileIndex: number;
-  startedAt: number;
-  endedAt: number;
 
   inverted?: boolean = false;
   leftHeavy?: boolean = false;
 
   depth = 0;
-  duration = 0;
   configSpace: Rect = new Rect(0, 0, 0, 0);
 
   formatter: (value: number) => string;
   frameIndex: Record<string, FlamegraphFrame> = {};
 
+  static Empty(): Flamegraph {
+    return new Flamegraph(Profile.Empty(), 0, {
+      inverted: false,
+      leftHeavy: false,
+    });
+  }
+
+  static From(from: Flamegraph, {inverted = false, leftHeavy = false}): Flamegraph {
+    return new Flamegraph(from.profile, from.profileIndex, {inverted, leftHeavy});
+  }
+
   constructor(
     profile: Profile,
     profileIndex: number,
-    {inverted = false, leftHeavy = false}: {inverted?: boolean; leftHeavy?: boolean} = {}
+    {inverted = false, leftHeavy = false}: {inverted?: boolean; leftHeavy?: boolean} = {},
+    configSpace?: Rect
```

Comment:
would it make more sense to include the config space setting inside the options object in the 3rd argument?